onadata/libs/tests/mixins/make_submission_mixin.py

```diff
@@ -0,0 +1,144 @@
+                                        password=auth.password)
+            self.response = None  # Reset in case error in viewset below
+            self.response = self.submission_view(request, username=username)
+            if auth and self.response.status_code == 401:
+                request.META.update(auth(request.META, self.response))
+                self.response = self.submission_view(request, username=username)
+
+        if forced_submission_time:
+            instance = Instance.objects.order_by('-pk').all()[0]
+            instance.date_created = forced_submission_time
+            instance.save()
+            instance.parsed_instance.save()
+
+        # remove temporary file if stored
+        if add_uuid:
+            os.unlink(path)
+
+    def _make_submission_w_attachment(self, path, attachment_path):
+
+        with open(path, 'rb') as f:
+            a = open(attachment_path, 'rb')
+            post_data = {'xml_submission_file': f, 'media_file': a}
+            url = f'/{self.user.username}/submission'
+            auth = DigestAuth('bob', 'bob')
+            self.factory = APIRequestFactory()
+            request = self.factory.post(url, post_data)
+            request.user = authenticate(
+                username='bob', password='bob'
+            )
+            self.response = self.submission_view(
+                request, username=self.user.username
+            )
+
+            if auth and self.response.status_code == 401:
+                request.META.update(auth(request.META, self.response))
+                self.response = self.submission_view(
+                    request, username=self.user.username
+                )
+
+    def _make_submissions(
+        self,
+        username=None,
+        add_uuid=False,
+        should_store=True,
+        auth=None,
+        module_directory=None,
+    ):
+        """
+        Make test fixture submissions to current xform.
+
+        :param username: submit under this username, default None.
+        :param add_uuid: add UUID to submission, default False.
+        :param should_store: should submissions be save, default True.
+        """
+
+        if not module_directory:
+            module_directory = self.this_directory
+
+        paths = [os.path.join(
+            module_directory, 'fixtures', 'transportation',
+            'instances', s, s + '.xml') for s in self.surveys]
+        pre_count = Instance.objects.count()
+
+        for path in paths:
+            self._make_submission(path, username, add_uuid, auth=auth)
+
+        post_count = (
+            pre_count + len(self.surveys) if should_store else pre_count
+        )
+
+        self.assertEqual(Instance.objects.count(), post_count)
+        self.assertEqual(self.xform.instances.count(), post_count)
+        xform = XForm.objects.get(pk=self.xform.pk)
+        self.assertEqual(xform.num_of_submissions, post_count)
+        self.assertEqual(xform.user.profile.num_of_submissions, post_count)
```

Comment:
@dorey would say `assert...`, @jnm would say `self.assertEqual(...)` .
Since most of the code is using `self.assertEqual` and we did not have a consensus, I kept the most used one.